frame/pablo/src/uniswap_tests.rs

```diff
@@ -428,3 +431,259 @@ proptest! {
 	})?;
 }
 }
+
+mod twap {
+	use super::*;
+	use crate::types::TimeWeightedAveragePrice;
+	use composable_traits::defi::Rate;
+	use sp_runtime::{
+		traits::{CheckedMul, One},
+		FixedPointNumber, SaturatedConversion,
+	};
+
+	fn run_to_block(n: BlockNumber) {
+		Pablo::on_finalize(System::block_number());
+		for b in (System::block_number() + 1)..=n {
+			next_block(b);
+			if b != n {
+				Pablo::on_finalize(System::block_number());
+			}
+		}
+	}
+
+	fn next_block(n: u64) {
+		Timestamp::set_timestamp(MILLISECS_PER_BLOCK * n);
+		System::set_block_number(n);
+		Pablo::on_initialize(n);
+	}
+
+	#[test]
+	fn twap_1() {
```

Comment:
Please describe what the test does. If I see in the output:

```
test::twap1 failed
```

vs

```
test::twap_with_two_assets
```
```

```
test::